src/domain/LegalFactDownloadMetadataRecord.ts

```diff
@@ -18,6 +18,9 @@ export type LegalFactDownloadMetadataRecord = AuditRecord & {
   output: Response<200, LegalFactDownloadMetadataResponse> | Response<403, UnauthorizedMessageBody> | Response<404>;
 };
 
+export const isLegalFactDownloadMetadataRecord = (record: Record): O.Option<LegalFactDownloadMetadataRecord> =>
+  record.type === 'LegalFactDownloadMetadataRecord' ? O.some(record) : O.none;
```

Comment:
```suggestion
  O.fromPredicate(record => record.type === 'LegalFactDownloadMetadataRecord')
```

Reply:
Actually, this doesn't work, because the compiler doesn't infer the proper type